api/src/main/java/com/spotify/spydra/api/DataprocAPI.java

```diff
@@ -86,7 +86,8 @@ public boolean deleteCluster(SpydraArgument arguments) throws IOException {
   public boolean submit(SpydraArgument arguments) throws IOException {
     boolean success = false;
     try {
-      success = gcloud.submit(arguments.getJobType(), arguments.getRegion(),
+      String pyFile = arguments.submit.pyFile.orElse(null);
```

Comment:
What is the reason to unpack the Optional here and then have a null check later on? Couldn't you simply pass the optional to gcloud.submit?

Reply:
Saw a message fly by my IDE, searched, found https://stackoverflow.com/questions/31922866/why-should-java-8s-optional-not-be-used-in-arguments. I prefer optionals too. :)